src/sentry/data_export/processors/discover.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import absolute_import
+
+import logging
+
+from sentry.api.utils import get_date_range_from_params
+from sentry.snuba import discover
+from sentry.models import Group, Project
+
+from ..base import ExportError
+
+logger = logging.getLogger(__name__)
+
+
+class DiscoverProcessor(object):
+    """
+    Processor for exports of discover data based on a provided query
+    """
+
+    def __init__(self, organization_id, discover_query):
+        self.projects = self.get_projects(organization_id, discover_query)
+        self.start, self.end = get_date_range_from_params(discover_query)
+        self.params = {
+            "organization_id": organization_id,
+            "project_id": [project.id for project in self.projects],
+            "start": self.start,
+            "end": self.end,
+        }
+        self.header_fields = discover_query["field"]
+        self.data_fn = self.get_data_fn(
+            fields=discover_query["field"], query=discover_query["query"], params=self.params
+        )
+
+    @staticmethod
+    def get_projects(organization_id, query):
+        project_ids = query.get("project")
```

Comment:
If you normalized `query.project` to a list before starting the export you could have simpler code here.

Reply:
Done!